### DISCUSSION TITLE
GC Status Bar

### DISCUSSION THREAD

**Discussion by numeralnathan** [dbeaver/dbeaver#9448] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Sometimes the JDBC driver throws an OutOfMemoryError.  I have to change the DBeaver configuration and increase the heap size.  I would like to be able to monitor the DBeaver heap usage from the UI so I can see if DBeaver is going to run out of memory.

**Describe the solution you'd like**
Please add a GC status bar at the bottom of DBeaver.  The status bar should show the current heap usage and the committed heap size.  The current heap usage should update every second.

There should also be a garbage can so I can force a GC.  This is useful for clearing the heap after viewing a large result.

When I mouse over the GC status bar, the tooltip should show me the current heap size, committed heap size and the maximum heap size.

Here is a screenshot of the GC status bar from Eclipse IDE.

![image](https://user-images.githubusercontent.com/1236594/89302957-c1ff6e80-d628-11ea-8c9d-915efcddcc2a.png)

**Describe alternatives you've considered**
I could open Visual VM or JConsole to monitor the heap usage and force GC.  This is cumbersome.

**Comment by kseniiaguzeeva** [comment, marked as answer]
It is already implemented. Please:
1. open Window>Preferences>General
2. enable **Show heap status**
![6LedvikbHu](https://user-images.githubusercontent.com/51405061/89635350-8cb88200-d8af-11ea-9f79-8d0265679f6e.png)
